homeassistant/components/systemmonitor/sensor.py

```diff
@@ -52,71 +57,226 @@
 else:
     CPU_ICON = "mdi:cpu-32-bit"
 
-SENSOR_TYPE_NAME = 0
-SENSOR_TYPE_UOM = 1
-SENSOR_TYPE_ICON = 2
-SENSOR_TYPE_DEVICE_CLASS = 3
-SENSOR_TYPE_MANDATORY_ARG = 4
-
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
-# Schema: [name, unit of measurement, icon, device class, flag if mandatory arg]
-SENSOR_TYPES: dict[str, tuple[str, str | None, str | None, str | None, bool]] = {
-    "disk_free": ("Disk free", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use": ("Disk use", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use_percent": (
-        "Disk use (percent)",
-        PERCENTAGE,
-        "mdi:harddisk",
-        None,
-        False,
+SENSOR_TYPES: Final[tuple[SystemMonitorSensorEntityDescription, ...]] = (
+    SystemMonitorSensorEntityDescription(
+        key="disk_free",
+        name="Disk free",
+        unit_of_measurement=DATA_GIBIBYTES,
+        icon="mdi:harddisk",
+        device_class=None,
+        state_class=STATE_CLASS_MEASUREMENT,
+        mandatory=False,
     ),
-    "ipv4_address": ("IPv4 address", "", "mdi:server-network", None, True),
-    "ipv6_address": ("IPv6 address", "", "mdi:server-network", None, True),
-    "last_boot": ("Last boot", None, None, DEVICE_CLASS_TIMESTAMP, False),
-    "load_15m": ("Load (15m)", " ", CPU_ICON, None, False),
-    "load_1m": ("Load (1m)", " ", CPU_ICON, None, False),
-    "load_5m": ("Load (5m)", " ", CPU_ICON, None, False),
-    "memory_free": ("Memory free", DATA_MEBIBYTES, "mdi:memory", None, False),
-    "memory_use": ("Memory use", DATA_MEBIBYTES, "mdi:memory", None, False),
-    "memory_use_percent": (
-        "Memory use (percent)",
-        PERCENTAGE,
-        "mdi:memory",
-        None,
-        False,
+    SystemMonitorSensorEntityDescription(
+        key="disk_use",
+        name="Disk use",
+        unit_of_measurement=DATA_GIBIBYTES,
+        icon="mdi:harddisk",
+        device_class=None,
+        state_class=STATE_CLASS_MEASUREMENT,
+        mandatory=False,
     ),
-    "network_in": ("Network in", DATA_MEBIBYTES, "mdi:server-network", None, True),
-    "network_out": ("Network out", DATA_MEBIBYTES, "mdi:server-network", None, True),
-    "packets_in": ("Packets in", " ", "mdi:server-network", None, True),
-    "packets_out": ("Packets out", " ", "mdi:server-network", None, True),
-    "throughput_network_in": (
-        "Network throughput in",
-        DATA_RATE_MEGABYTES_PER_SECOND,
-        "mdi:server-network",
-        None,
-        True,
+    SystemMonitorSensorEntityDescription(
+        key="disk_use_percent",
+        name="Disk use (percent)",
+        unit_of_measurement=PERCENTAGE,
+        icon="mdi:harddisk",
+        device_class=None,
```

Comment:
This default to `None` and should not be set here+

(Same goes for the others, but I'll stop that comment here)